src/sidebar/services/groups.js

```diff
@@ -163,164 +163,144 @@ function groups(
    *
    * @return {Promise<Group[]>}
    */
-  function load() {
-    let uri = Promise.resolve(null);
+  async function load() {
+    // Step 1: Get the URI of the active document, so we can fetch groups
+    // associated with that document.
+    let uri = null;
     if (isSidebar) {
-      uri = getDocumentUriForGroupSearch();
+      uri = await getDocumentUriForGroupSearch();
     }
-    const directLinkedGroupId = store.getState().directLinkedGroupId;
-    const directLinkedAnnId = store.getState().directLinkedAnnotationId;
+    documentUri = uri;
+
+    // Step 2: Concurrently fetch the groups the user is a member of,
+    // the groups associated with the current document and the annotation
+    // or group that was direct-linked (if any).
```

Comment:
It doesn't make sense to allow a direct link to an annotation AND a group, because the client can only follow one or the two instructions. The fact that a user can specify both in the client config, now that I reconsider it, a flaw in the API design. If we had designed it like this:

```js
{
  // Direct link to an annnotation...
  directLink: { 'type': 'annotation', id: '123' }, 

  // ...OR direct link to a group
  directLink: { 'type': 'group', id: '456' }, 
}
```

That would have eliminated that possibility.

Given that we've shipped this, I think we should simply decide that if both settings are present, one takes precedence over the other.

Reply:
Here's the thread on slack where I considered what to do here: https://hypothes-is.slack.com/archives/C1M8NH76X/p1554397519016100. In summary, the direct-linked annotation does take precedents but it still may grab the direct-linked group that wasn't in the list of groups and add it to the list.

Reply:
While I do like that suggested approach/change, I do think this comment still needs to be adjusted to accurately describe what's _currently_ happening.

Reply:
@robertknight This was the comment I was referring to that needs addressing.